using-timescaledb/distributed-hypertables.md

```diff
@@ -103,6 +103,90 @@ the data node's data is outside the retention window,
+to that many chunks.
+
+With a replication factor of one, data is written to the data nodes
+using a one-phase commit, which basically means that every transaction
+on the access node opens a similar transaction on the data
+node. Committing the transaction involves sending a commit message to
+the data node and no additional coordination.
+
+With a higher replication factor, data will be written to the chunks
+using two-phase commit. This protocol is more complicated, but it
+guarantees that the chunks are kept in sync on all data nodes and that
+failure to write to one of the data nodes will generate an error.
+
+To create a hypertable with a replication factor higher than 1, you
+can use the `replication_factor` parameter to
+[`create_distributed_hypertable`][create_distributed_hypertable]. For
+example, if you want to create a hypertable where each chunk has three
+replicas:
+
+```sql
+SELECT create_distributed_hypertable('conditions', 'time', 'location',
+	replication_factor => 3);
+```
+
+If you already have a hypertable and instead want to change the
+replication factor, you can use the function
+[`set_replication_factor`][set_replication_factor].
+
+```sql
+SELECT set_replication_factor('conditions', 3);
+```
+
+Having a replication factor larger than one is a simple way to ensure
+that you have redundancy of the data and that a crash of a data node
+will not cause you to lose the data. This, however, comes at the cost
+of an increased average commit time and more data to manage.
+
+Transactions that write to the distributed table will be slower on
+average with multiple replicas (when compared with a single replica)
+for two reasons:
+- The two-phase commit procedure induces additional processing
+  compared to one-phase commit. This is, however, minor and does not
+  significantly affect the commit time.
+- Since there are now more transactions to be committed, the commit
+  time of the two-phase commit (also known as the *global commit*) is
+  the maximum time of the commits on the data nodes (also known as
+  *local commits*), which on average is going to be higher than the
+  average of the commit of each transaction on the data nodes. (The
+  proof of this is left as an excercies for the reader.)
+
+When executing a two-phase commit, the commit will block until it
+receives an acknowledgement that the transaction is committed on all
+data nodes, or a connection times out, whichever comes first. This
+means that if one of the data nodes crash during a commit (or is
+unresponsive for some other reason) the connection will eventually
+time out and abort the transaction.
+
+Read-only query execution is not affected by the redundant data: the
+query will just be split up and send the sub-query to *one* of the
+replicas (not all of them), so the replication factor does not affect
+query execution time in any significant manner.
+
+#### Limitations
+
+The two-phase commit protocol is stable and handles failures without
```

Comment:
There seems to be a fundamental misunderstanding here: two phase commit has no connection to replication factor.

Also, I think this section is misleading. AFAIK, two-phase commit only guarantees transactional semantics (atomicity) but doesn't "handle" failures (at least not in any sense of resolving them). It only ensures that, in case of a failure, a distributed operation fails globally and not only locally on some nodes.

Reply:
I was just tired and got into my head that single replica == single node, which is not the case. I used "handle" in a loose sense that it would ensure transactional semantics, not that it would resolve the problem, but I reworded to avoid the confusion.